x-pack/plugins/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -236,20 +237,20 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
  */
 const transformToString = () => {
   let lineBuffer: string = '';
+  const decoder = new TextDecoder();
+  const encoder = new TextEncoder();
 
   return new Transform({
     transform(chunk, encoding, callback) {
-      const decoder = new TextDecoder();
-      const encoder = new TextEncoder();
-      const lines = decoder.decode(chunk).split('\n');
+      const chunks = decoder.decode(chunk);
+      const lines = chunks.split('\n');
+      if (lines[lines.length - 1] === '') {
```

Comment:
@dgieselaar If I do not do this, there is an extra empty line and the `correctly buffers partial lines` test fails